core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1085,6 +1085,10 @@ class DAGScheduler(
 
         if (disallowStageRetryForTest) {
           abortStage(failedStage, "Fetch failure will not retry stage due to testing config")
+        } else if (failedStage.failAndShouldAbort()) {
```

Comment:
It looks like this will mark the stage as failed each time a single task in the stage fails? So if 4 tasks in the stage fail with a fetch failure, the stage will be aborted? Am I misunderstanding this?